rules/readfile.go

```diff
@@ -0,0 +1,50 @@
+// (c) Copyright 2016 Hewlett Packard Enterprise Development LP
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rules
+
+import (
+	"go/ast"
+	"go/types"
+
+	"github.com/GoASTScanner/gas"
+)
+
+type readfile struct {
+	gas.CallList
+}
+
+// Match inspects AST nodes to determine if the match the methods `os.Open` or `ioutil.ReadFile`
```

Comment:
Maybe it makes sense to catch the entire family of function which can modify files, such as os.Create, os.Truncate, os.Rename, os.Remove ...

Reply:
Yeah, I was thinking this also. But it might make sense to separate out vulns wrt Read and Write. However both read/write follow the same vuln pattern (e.g., tainted input going to the sensitive function)